diff --git a/src/zimg/api/zimg.h b/src/zimg/api/zimg.h
index 1a4fb14..5e4fe0b 100644
--- a/src/zimg/api/zimg.h
+++ b/src/zimg/api/zimg.h
@@ -5,7 +5,7 @@

 /* Support for ELF hidden visibility. DLL targets use export maps instead. */
 #if defined(_WIN32) || defined(__CYGWIN__)
-  #define ZIMG_VISIBILITY
+  #define ZIMG_VISIBILITY __declspec(dllexport)
 #elif defined(__GNUC__)
   #define ZIMG_VISIBILITY __attribute__((visibility("default")))
 #else
